bundles/com.zeligsoft.cx.codegen.ui/src/com/zeligsoft/cx/codegen/ui/preferences/CodegenReportPreferencePage.java

```diff
@@ -0,0 +1,243 @@
+	 * .swt.widgets.Composite)
+	 */
+	@Override
+	protected Control createContents(Composite parent) {
+
+		final Composite panel = new Composite(parent, SWT.NONE);
+
+		final GridLayout layout = new GridLayout(1, false);
+		panel.setLayout(layout);
+
+		final GridData data = new GridData(SWT.FILL, SWT.FILL, true, false, 1, 1);
+		panel.setLayoutData(data);
+
+		createFileCollectorGroup(panel);
+
+		panel.addControlListener(new ControlAdapter() {
+			public void controlResized(ControlEvent e) {
+				data.widthHint = panel.getClientArea().width - 2 * layout.marginWidth;
+				panel.layout(true);
+			}
+		});
+		panel.layout(true);
+		return panel;
+	}
+
+	private void createFileCollectorGroup(Composite panel) {
+		Group group = new Group(panel, SWT.NULL);
+		group.setLayout(new GridLayout(1, false));
+		final GridData data = new GridData(SWT.FILL, SWT.TOP, true, false, 1, 1);
+		group.setLayoutData(data);
+		group.setText(Messages.CXCodegenPreferencePage_FileCollectorGroup);
+
+		createFileCollectorQuestion(group);
+		createFilesSetCheckbox(group, FilesSet.ADDED);
+		createFilesSetCheckbox(group, FilesSet.CHANGED);
+		createFilesSetCheckbox(group, FilesSet.REMOVED);
+		createFilesSetCheckbox(group, FilesSet.UNCHANGED);
+		createFilesSetCheckbox(group, FilesSet.BEFORE);
+		createFilesSetCheckbox(group, FilesSet.AFTER);
+		group.layout(true);
+	}
+
+	private void createFileCollectorQuestion(Composite group) {
+		Label label = new Label(group, SWT.WRAP | SWT.LEFT);
+		final GridData data = new GridData(SWT.FILL, SWT.TOP, true, false, 1, 1);
+		data.widthHint = 400;
+		label.setLayoutData(data);
+
+		label.setText(Messages.CXCodegenPreferencePage_FileCollectorQuestion);
+	}
+
+	private void createFilesSetCheckbox(Composite group, FilesSet filesSet) {
+		includeFilesSetCheckbox.put(filesSet, new Button(group, SWT.CHECK));
+		includeFilesSetCheckbox.get(filesSet).setText(filesSetCheckboxLabel.get(filesSet));
+		includeFilesSetCheckbox.get(filesSet).setSelection(PREFERENCE_SERVICE.getBoolean(CXActivator.PLUGIN_ID,
+				checkBoxPreferenceKey.get(filesSet), checkBoxPreferenceDefault.get(filesSet), null));
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.eclipse.jface.preference.PreferencePage#performApply()
+	 */
+	@Override
+	protected void performApply() {
+		for (FilesSet filesSet : FilesSet.values()) {
+			boolean checkBoxValue = includeFilesSetCheckbox.get(filesSet).getSelection();
+			WORKSPACE_PREFERENCES.putBoolean(checkBoxPreferenceKey.get(filesSet), checkBoxValue);
+		}
+	};
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.eclipse.jface.preference.PreferencePage#performCancel()
+	 */
+	@Override
+	public boolean performCancel() {
+		Activator.getDefault().info("Cancelling change to preference. ");
+		for (FilesSet filesSet : FilesSet.values()) {
+			WORKSPACE_PREFERENCES.putBoolean(checkBoxPreferenceKey.get(filesSet), currentValue.get(filesSet));
+		}
+		try {
+			WORKSPACE_PREFERENCES.flush();
```

Comment:
I don't think we should be flushing anything for "cancel" operation. or am I missing something??